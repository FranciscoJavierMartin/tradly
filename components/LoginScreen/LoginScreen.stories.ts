import LoginScreen from './LoginScreen.vue';

export default {
  title: 'LoginScreen',
  component: LoginScreen,
  args: {},
  parameters: {
    layout: 'fullscreen',
  },
};

const Template = (args) => ({
  components: { LoginScreen },
  setup() {
    return { args };
  },
  template: '<LoginScreen v-bind="args"/>',
});

export const Basic = Template.bind({});
Basic.args = {};
