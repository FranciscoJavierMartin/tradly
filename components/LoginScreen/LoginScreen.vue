<template>
  <div class="c-login-screen">
    <h1>Welcome to tradly</h1>
    <span>Login to your account</span>
    <form>
      <input type="email" placeholder="Email" />
      <input type="password" placeholder="Password" />
      <CustomButton text="Login" inverted />
    </form>
    <a>Forgot you password?</a>
    <span>Don't have an account?</span> <a>Sign up</a>
  </div>
</template>

<script setup lang="ts">
import CustomButton from '../CustomButton/CustomButton.vue';
const props = defineProps({});
</script>

<style lang="scss">
@import '../../assets/styles/main.scss';

.c-login-screen {
  background-color: $dark-green;
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  text-align: center;
  color: $white;

  form {
    display: flex;
    flex-direction: column;
    padding: 20px;

    input {
      color: $white;
      background-color: $dark-green;
      border: 1px solid $white;
      border-radius: 24px;
      font-family: $monserrat-font;
      font-weight: normal;
      line-height: 22px;
      padding: 10px;

      &::placeholder {
        color: $white;
        background-color: $dark-green;
      }
    }
  }
}
</style>
