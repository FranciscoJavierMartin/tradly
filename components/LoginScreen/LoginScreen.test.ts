import '@testing-library/jest-dom';
import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/vue';
import LoginScreen from './LoginScreen.vue';

describe('LoginScreen.vue', () => {
  test('should render', () => {
    const { container } = render(LoginScreen);

    expect(
      screen.getByRole('heading', { name: /Welcome to tradly/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
