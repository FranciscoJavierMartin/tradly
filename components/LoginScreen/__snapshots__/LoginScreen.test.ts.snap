// Vitest Snapshot v1

exports[`LoginScreen.vue > should render 1`] = `
<div
  class="c-login-screen"
>
  <h1>
    Welcome to tradly
  </h1>
  <span>
    Login to your account
  </span>
  <form>
    <input
      placeholder="Email"
      type="email"
    />
    <input
      placeholder="Password"
      type="password"
    />
    <button
      class="c-custom-button c-custom-button--inverted"
    >
      Login
    </button>
  </form>
  <a>
    Forgot you password?
  </a>
  <span>
    Don't have an account?
  </span>
   
  <a>
    Sign up
  </a>
</div>
`;
